tests/test_edgeql_scope.py

```diff
@@ -1097,16 +210,28 @@ async def test_edgeql_scope_binding_02a(self):
                 name := {'Alice', 'Bob'},
                 L := (name, (
                     SELECT User
-                    FILTER .name = name
```

Comment:
Ok, so `.name` and `name` shouldn't have been factored even before this change.

Why did this work before? Had we allowed FILTER condition to be a set and applied `any` implicitly?

Reply:
`name` was factored with that use of `name` in the tuple above

But also we *do* allow `FILTER` conditions to be a set and apply `any` implicitly, though in 6.0 we now emit a warning; I don't think is happening here, though, since it name is getting factored

Reply:
Oh, I missed the `(name, (`, right.